### PR TITLE
finder log format

### PR DESCRIPTION
**Pull request by SimonSK**
Most of the case-study log message related changes are reverted.
Added some functions to log structs following a consistent format.

### CODE REVIEW

File: eth/peer.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (p *peer) SendTransactions(txs types.Transactions) error {
 	for _, tx := range txs {
 		p.knownTxs.Add(tx.Hash())
 	}
-	return p2p.SendEthSubproto(p.rw, TxMsg, txs, p.ID())
```
Comment (zzma) [inline eth/peer.go:139]: Why did you change this function name? Seems like the old name is more specific

File: p2p/discover/node.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ func (n NodeID) String() string {
 
 // The Go syntax representation of a NodeID is a call to HexID.
 func (n NodeID) GoString() string {
-	return fmt.Sprintf("%x", n[:])
+	return fmt.Sprintf("discover.HexID(\"%x\")", n[:])
```
Comment (zzma) [inline p2p/discover/node.go:240]: what is this for?

File: p2p/node_info.go [hunk 3]
```diff
@@ -79,7 +80,67 @@ type Info struct {
...
+			} else {
+				s += fmt.Sprintf("%s:nil", elemName)
+			}
+		case reflect.Struct:
+			continue
+		default:
+			s += fmt.Sprintf("%s:%v", elemName, elem.Interface())
+		}
+	}
+	return s
```
Comment (zzma) [inline p2p/node_info.go:111]: Can you provide an example of a sample output?

File: eth/peer.go (hunk 1 above)
Comment (zzma) [reply eth/peer.go:139]: Actually it looks like you are using a different function altogether.

File: p2p/message.go [hunk 4]
```diff
@@ -103,70 +93,12 @@ type MsgReadWriter interface {
...
-	0x05: empty, //GetBlockBodies
-	0x06: empty, //BlockBodies
-	0x07: empty, //NewBlockMsg
-	0x0d: empty, //GetNodeData
-	0x0e: empty, //NodeData
-	0x0f: empty, //GetReceipts
-	0x10: empty, //Receipts
-}
-
-func SendEthSubproto(w MsgWriter, msgcode uint64, data interface{}, peers ...discover.NodeID) error {
```
Comment (zzma) [inline p2p/message.go:127]: Why do we no longer need to call this function?

File: p2p/message.go [hunk 5]
```diff
@@ -103,70 +93,12 @@ type MsgReadWriter interface {
...
-	if _, ok := dataExcludedMsgs[msgcode]; ok {
-		data = "<OMITTED>"
-	}
-
-	unixTime := float64(time.Now().UnixNano()) / 1000000000
-	log.Proto(">>"+msgType, "sentAt", unixTime, "obj", data, "size", int(size), "peer", peer)
-	return w.WriteMsg(Msg{Code: msgcode, Size: uint32(size), Payload: r})
-}
-
-func SendDEVp2p(w MsgWriter, msgcode uint64, data interface{}, peers ...discover.NodeID) error {
```
Comment (zzma) [inline p2p/message.go]: Same - why do we no longer need to call this function, and call Send instead?

**Review by zzma — COMMENTED** [review]
Oops forgot to actually submit the review again. Left some questions.

File: eth/peer.go (hunk 1 above)
Comment (SimonSK) [reply eth/peer.go:139]: I just reverted to what master branch had.
I think `send/SendItems` was split into those 2 functions so that we have better control over log messages.
Since we aren't logging those anymore, I just switched them back.

File: p2p/discover/node.go (hunk 2 above)
Comment (SimonSK) [reply p2p/discover/node.go:240]: That's what master branch had.
I don't think it really affects our log messages as `String()` is used by default in most cases.

File: p2p/message.go (hunk 4 above)
Comment (SimonSK) [reply p2p/message.go:127]: Answered above.

File: p2p/message.go (hunk 5 above)
Comment (SimonSK) [reply p2p/message.go]: Answered above.

File: p2p/node_info.go (hunk 3 above)
Comment (SimonSK) [reply p2p/node_info.go:111]: `RowId:826 Keccak256Hash:214ce55662b500351067ef9e9540a3f1f5d5481f6fba27bac099e20bd3169fbb IP:45.32.118.170 TCPPort:27030 RemotePort:27030 P2PVersion:5 ClientId:Geth/lth/v1.0.8-inviting-version-5cf75a30/linux-amd64/go1.9.2 Caps:eth/62,eth/63 ListenPort:0 FirstHelloAt:1518874814.916871 LastHelloAt:1519005739.049556 ProtocolVersion:63 NetworkId:1 FirstReceivedTd:221514441192 LastReceivedTd:223257459396 BestHash:5afbf82e4324a83889b2286a983c1a7813d76a15dc5b7437c5a677ee06d0c97f GenesisHash:647d1c9a396e07f64a808533538e39973d8c7955d5d201486bd4470d17f60033 FirstStatusAt:1518874815.160252 LastStatusAt:1518880227.140020 DAOForkSupport:0`

The full message looks like this:
`DEBUG|2018-02-18T20:02:19.055127017-0600|1519005739.055127|Executed UpdateNodeInfo sql statement|id=a6b5ced2b6aa174de748b82b770b27e58a9f71c212eaf778a0cba50722bebab8d99e9d69488d39a8a9ebcd9d84f12a0d6968d4b58d594a5b2cb6a20d89e63e65|newInfo="RowId:826 Keccak256Hash:214ce55662b500351067ef9e9540a3f1f5d5481f6fba27bac099e20bd3169fbb IP:45.32.118.170 TCPPort:27030 RemotePort:27030 P2PVersion:5 ClientId:Geth/lth/v1.0.8-inviting-version-5cf75a30/linux-amd64/go1.9.2 Caps:eth/62,eth/63 ListenPort:0 FirstHelloAt:1518874814.916871 LastHelloAt:1519005739.049556 ProtocolVersion:63 NetworkId:1 FirstReceivedTd:221514441192 LastReceivedTd:223257459396 BestHash:5afbf82e4324a83889b2286a983c1a7813d76a15dc5b7437c5a677ee06d0c97f GenesisHash:647d1c9a396e07f64a808533538e39973d8c7955d5d201486bd4470d17f60033 FirstStatusAt:1518874815.160252 LastStatusAt:1518880227.140020 DAOForkSupport:0"`

Comment (zzma) [reply p2p/node_info.go:111]: Is using space as a delimiter in newInfo a potential issue? or are spaces not allowed in any of the fields?

**Review by zzma — APPROVED** [review]
minor comment, otherwise LGTM

File: p2p/node_info.go (hunk 3 above)
Comment (SimonSK) [reply p2p/node_info.go:111]: Only fields that could have characters outside of alphabets and digits are `ClientId` and `Caps`. Other fields should have only numbers or hex values (or IPs).
All quotation marks, |, and whitespaces are removed from those two fields using `strings.Replacer` in line 258.
The Replacer object is initialized in `p2p/server.go:394`.
